django/views/csrf.py

```diff
@@ -61,6 +61,16 @@ def csrf_failure(request, reason="", template_name=CSRF_FAILURE_TEMPLATE_NAME):
             "re-enable them, at least for this site, or for “same-origin” "
             "requests."
         ),
+        "bad_origin": reason.startswith("Origin checking failed"),
+        "forwarded_may_fix": (
+            request.headers.get("X-Forwarded-Proto", "") == "https"
+            and not request.is_secure()
+            or request.headers.get("Origin", "").endswith(
+                f'://{request.headers.get("X-Forwarded-Host", "")}'
+            )
+        ),
```

Comment:
No tests fail if I was to update to:
```suggestion
        "forwarded_may_fix": True,
```
Can you write a test for the `False` case (probably asserts that `If the expected server origin looks correct, you may wish to add...` in the template).